content/the-basics-of-go.md

```diff
@@ -0,0 +1,865 @@
+
+  // Or a shorthand, which will be mostly used.
+  str := "Something fishy"
+
+  // You can also declare multiple variables at once.
+
+  var (
+    name string = "John"
+    age int = 30
+  )
+
+  // Or..
+
+  var name, age = "John", 30
+  name, age := "John", 30 // Shorthand declaration.
+}
+```
+
+Go has something called **zero value**. Zero value is the default value of a variable. For example, if you declare a
+string type variable, it will have an empty string ("") as the default value. If you declare an integer type variable,
+it will have 0 as the default value. And if you declare a boolean type variable, it will have false as the default value.
+
+By default, variables in Go are mutable, meaning you can change the value of it after the declaration. So this is valid:
+
+```go
+package main
+
+func main() {
+  var str = "Something fishy"
+  str = "Something else"
+  str = "Oh, I also thought of something else again"
+}
+```
+
+What if you want it to be immutable? We'll need to use the keyword...
+
+## Constant
+
+Unlike variables that are mutable, constants are immutable. You can't change the value of it after its' declaration.
+
+```go
+package main
+
+const (
+  pi float64 = 3.14
+  e  float64 = 2.71
+)
+
+const NUMBER_OF_LIVES = 9
+
+// Yes, you can declare a constant or even a variable outside a function.
+// It's valid and nothing's wrong with it.
+```
+
+## Array, slices and maps
+
+Array is simple. It's a fixed-size list of elements of the same type.
+
+```go
+var people [3]string = [3]string{"John", "Paul", "George"}
+// But GoLand, the Go IDE by JetBrains would find that to be redundant,
+// so you'll have to remove the [3]string type.
+var people = [3]string{"John", "Paul", "George"}
+
+fmt.Println(people[1]) // Paul
+```
+
+Slices are like array, but it's a variable-size list of elements of the same type.
+
+```go
+var people = []string{"John", "Paul", "George"}
+
+fmt.Println(people[2]) // George
+```
+
+Well, what's the difference between array and slices? Simple. They have different capacity size.
+If you have an array of string that have a length of 10, you can't add another value on the 10-th index
+(as Go array starts from 0). But if you have a slice of string that have a length of 10, you can add
+another value on the 10-th index.
+
+Now, you've talked about add another value to a slice. How do we do that? Use `append`.
+To know the length of an array? Use `len`. To know the capacity of an array? Use `cap`.
+
+```go
+var people = []string{"John", "Paul", "George"}
+people = append(people, "Ringo")
+
+fmt.Println(people) // ["John" "Paul" "George" "Ringo"]
+
+var length = len(people) // 4
+var capacity = cap(people) // 6
+```
+
+Wait, why the capacity of people is 6? Well, I'm not going to explain this right now. But if you're
+curious, you can read [this article](https://go.dev/blog/slices-intro#TOC_5.) to know more.
+
+Maps is like a Dictionary, Vector, Object, or Hash in other languages.
```

Comment:
I don't think it's similar to Vector. Vector is more similar to Go's slices, a dynamic/resizable/growable array. 

See:
- https://doc.rust-lang.org/std/vec/struct.Vec.html
- https://en.cppreference.com/w/cpp/container/vector

Reply:
I'm not talking about Rust or C++ lol, but ok, this might be a bit misleading